src/Agent/NewRelic/Profiler/Configuration/Configuration.h

```diff
@@ -423,9 +423,9 @@ namespace NewRelic { namespace Profiler { namespace Configuration {
             return false;
         }
 
-        bool IsW3wpProcess(const xstring_t& processName)
+        bool IsW3wpProcess(const xstring_t& processName, xstring_t const& parentProcessName)
         {
-            return Strings::EndsWith(processName, _X("W3WP.EXE"));
+            return Strings::EndsWith(processName, _X("W3WP.EXE")) || Strings::EndsWith(parentProcessName, _X("W3WP.EXE"));
```

Comment:
I realize it was already like this, but just doing an `EndsWith` could theoretically result in a false positive. Thoughts?

Reply:
I suppose one could get clever and parse the path down to just the filename. Not sure it's worth the effort, though. 

I did wonder, however, if I should be doing (pseudocode)
```
processName == "W3WP.EXE" || parentProcessName == "W3WP.EXE
```
 or if it would be better to do  
```
processName == "DOTNET.EXE" && parentProcessName == "W3WP.EXE"
```

Reply:
What if we log it as `process X with parent Y is/is not IIS` Then if something goes wrong we'll at least know why.